sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -131,8 +145,92 @@ export class ApplicationController extends BaseController {
         "Error while associating the assessment workflow.",
       );
     }
+  }
+
+  /**
+   * Creates a new application draft for the authenticated student.
+   * The student is allowed to have only one draft application, so
+   * this method will create the draft or throw an exception in case
+   * of the draft already exists.
+   * @param payload payload to create the draft application.
+   * @param userToken token from the authenticated student.
+   * @returns the application id of the created draft or an
+   * HTTP exception if it is not possible to create it.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Post("draft")
+  async createDraftApplication(
+    @Body() payload: SaveApplicationDto,
+    @UserToken() userToken: IUserToken,
+  ): Promise<number> {
+    const programYear = await this.programYearService.getActiveProgramYear(
+      payload.programYearId,
+    );
+
+    if (!programYear) {
+      throw new UnprocessableEntityException(
+        "Program Year is not active, not able to create a draft application.",
+      );
+    }
+
+    const student = await this.studentService.getStudentByUserId(
+      userToken.userId,
+    );
+
+    try {
+      const draftApplication =
+        await this.applicationService.saveDraftApplication(
+          student.id,
+          payload.programYearId,
+          payload.data,
+          payload.associatedFiles,
+        );
+      return draftApplication.id;
+    } catch (error) {
+      if (error.name === MORE_THAN_ONE_APPLICATION_DRAFT_ERROR) {
+        throw new UnprocessableEntityException(
+          new ApiProcessError(error.message, error.name),
+        );
+      }
+      throw new InternalServerErrorException(
+        "Unexpected error while creating the draft application.",
+      );
+    }
+  }
+
+  /**
+   * Updates an existing application draft
+   * @param payload payload to update the draft application.
+   * @param applicationId draft application id.
+   * @param userToken token from the authenticated student.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/draft")
```

Comment:
IMO, @Patch("/draft/:applicationId") is more preferred REST endpoint.

Reply:
The resource here would be "api/application/:id", that's why the id is right beside the application. Draft and Submit would be a different way to persist the same resource, in this case, "the application". I don't mind changing if the team agrees to follow it.
@ann-aot @guru-aot, any thoughts on this?

When you mention "preferred REST endpoint" can you please elaborate on why it would be "preferred"?

Reply:
Oh I get it. Here Application is looked as resource not draft.